cmd/drone-server/drone.go

```diff
@@ -236,6 +244,14 @@ func main() {
 		auth.POST("", server.GetLogin)
 	}
 
+	redirects := r.Group("/redirect")
+	{
+		redirects.Use(server.SetDatastore(store))
+		redirects.Use(server.SetRepo())
+
+		redirects.GET("/:owner/:name/commit/:sha", server.RedirectSha)
```

Comment:
Ok. GitLab doesn't provide any method to set merge request status from CI, gitlab know about service ( url patterns, sended data ), and try to fetch build status. After status has been fetched, gitlab shows build status.

![screen shot 2015-08-24 at 21 43 15](https://cloud.githubusercontent.com/assets/683590/9449111/30cf125e-4aa9-11e5-8abe-4f1824682444.png)

As we see, we have a link "View build details", after click we need to drone build page ( but 0.4.0 not uses SHA ). After click we go to redirect handler, and handler searches commit with sha and branch in repos scope, and finally build find, we redirect user to frontend build page.